integration/spark/app/src/main/java/io/openlineage/spark/agent/lifecycle/RddExecutionContext.java

```diff
@@ -300,9 +311,25 @@ protected OpenLineage.Job buildJob(int jobId) {
             .getOverriddenAppName()
             .orElse(sparkContextOption.map(SparkContext::appName).orElse("unknown"));
     String jobName = name + "." + suffix;
-    return new OpenLineage.JobBuilder()
+
+    return openLineage
+        .newJobBuilder()
         .namespace(eventEmitter.getJobNamespace())
         .name(jobName.replaceAll(CAMEL_TO_SNAKE_CASE, "_$1").toLowerCase(Locale.ROOT))
+        .facets(
+            openLineage
+                .newJobFacetsBuilder()
+                .jobType(
+                    openLineage
+                        .newJobTypeJobFacetBuilder()
+                        .jobType(SPARK_JOB_TYPE)
+                        .processingType(
+                            StreamingContextUtils.hasActiveStreamingContext()
+                                ? SPARK_PROCESSING_TYPE_STREAMING
+                                : SPARK_PROCESSING_TYPE_BATCH)
+                        .integration("SPARK")
```

Comment:
I can't help but think that the `processing_engine` facet already captures this information, and that we're duplicating it here (albeit a small amount of duplication).

Reply:
You're right  as `integration=spark` is a duplicate. However, in this PR I am not modifying the facet definition, but just including facet for the RDD execution context where it's missing.